lib/opentelemetry/trace/exporter/http_client.lua

```diff
@@ -28,29 +33,52 @@ function _M.new(address, timeout, headers)
         uri = uri,
         timeout = timeout,
         headers = headers,
+        httpc = httpc,
     }
     return setmetatable(self, mt)
 end
 
-function _M.do_request(self, body)
-    local httpc = http.new()
-    httpc:set_timeout(self.timeout * 1000)
+function _M.do_request(self, body, encode_gzip)
+    self.httpc = self.httpc or http.new()
 
-    local res, err = httpc:request_uri(self.uri, {
+    encode_gzip = encode_gzip or false
+    self.httpc:set_timeout(self.timeout * 1000)
+
+    if encode_gzip then
+        -- Compress (deflate) request body
+        -- the compression should be set to Best Compression and window size
+        -- should be set to 15+16, see reference below:
+        -- https://github.com/brimworks/lua-zlib/issues/4#issuecomment-26383801
+        self.headers["Content-Encoding"] = "gzip"
+        local deflate_stream = zlib.deflate(zlib.BEST_COMPRESSION, 15+16)
+        local compressed_body, _, bytes_in, bytes_out = deflate_stream(body, "finish")
+
+        otel_global.metrics_reporter:record_value(
```

Comment:
Do we have reason to believe that `bytes_in` will differ from `string.len(body)`?

Reply:
see https://github.com/yangxikun/opentelemetry-lua/pull/87#discussion_r1481467320, they should be the same, I don't have a preference, I can change to use `string.len` if you prefer :+1:

Reply:
I think `string.len` in both places makes sense, for consistency's sake...

Reply:
you got it :+1: